docs/development.md

```diff
@@ -0,0 +1,74 @@
+# Setting Up a Development Environment
+
+
+## Table of contents
+
+1. [Installing the Zendesk App Tools (ZAT)](#dev-zat)
+2. [Setting up the Python environment](#dev-python)
+3. [Lauching the Flask app](#dev-flask)
+4. [Getting browser connected to the Zendesk instance](#dev-instance)
+
+
+## Installing the Zendesk App Tools (ZAT) <a name='dev-zat'></a>
+ZAT is used for developing Zendesk apps and it bridges communications between your local (or remote) app and your Zendesk product instance.
+
+To get it running, first you must install Ruby since ZAT is a Ruby gem.
+
+Then, install `rake` a build automation tool.
+```sh
+gem install rake
+```
+
+Next, install ZAT
+```sh
+gem install zendesk_apps_tools
+```
+
+Navigate to **app_local** directory it contains all necessary files for any ZAT commands to run. Here, start a local HTTP server and then you can talk to your Zendesk product instance.
+``` sh
+zat server
+```
+
+If you want to customise the ZAT service, use **manifest.json** file. This file specifies one or more locations in one or more Zendesk products. In this project, we will build a Support app served at `localhost:8080/sidebar`.
+```
+...
+
+  "location": {
+    "support": {
+      "ticket_sidebar": "http://127.0.0.1:8080/sidebar"
+    }
+  },
+
+...
+
+```
+
+## Setting up the Python environment <a name='dev-python'></a>
+Install Python 3.7 and pip install `poetry` package. `poetry` is a dependency management tool.
+```
+pip install --user poetry
+```
+Install ML libraries dependencies and Flask using `poetry`. Make sure the installation happens in the root directory.
+```sh
+poetry install
+```
+Activate the virtual environment.
+```
+poetry shell
```

Comment:
are we using a new dependency management tool here and not pipenv anymore? is it better?

Reply:
It's just for this repo. I couldn't install `tensorflow` on my machine with `pipenv` when `pytorch` was there, it took forever to lock. I got switched to `poetry` and so far the locking and dependency resolution, I feel, are much better than `pipenv`.

Just took a look at `pipenv` repo, I am glad they start putting new releases. For the past one and half a years no updates made this repo which made me thought it's a dead project.